test/tests/CommandPortTest.cpp

```diff
@@ -0,0 +1,43 @@
+#include <libstuff/SData.h>
+#include <test/lib/BedrockTester.h>
+
+struct CommandPortTest : tpunit::TestFixture {
+    CommandPortTest()
+        : tpunit::TestFixture("CommandPort", TEST(CommandPortTest::test)) { }
+
+    void test() {
+        BedrockTester tester;
+
+        // When we close the command port with a reason
+        SData closeCommandPort("SuppressCommandPort");
+        closeCommandPort["reason"] = "testCommandPort";
+        string response = tester.executeWaitMultipleData({closeCommandPort})[0].content;
+
+        // The status command should show it in commandPortBlockReasons
+        SData status("Status");
+        response = tester.executeWaitMultipleData({status}, 10, true)[0].content;
+        ASSERT_TRUE(SContains(response, "commandPortBlockReasons"));
```

Comment:
You should be able to use `SParseJSONObject` on `response` and then `SParseJSONArray` on the correct field, something like:

`list<string> reasons = SParseJSONArray(SParseJSONObject(response)["commandPortBlockReasons"]);`

And then check for the reason in the list (this applies to later checks as well)